### DISCUSSION TITLE
different type into_response() return error silently

### DISCUSSION THREAD

**Discussion by freedit-dev** [tokio-rs/axum#2038] [Q&A] [answered]
- [x] I have looked for existing issues (including closed) about this

### Version

```
├── axum v0.6.18
│   ├── axum-core v0.3.4
│   ├── axum-macros v0.3.7 (proc-macro)
│   ├── axum v0.6.18 (*)
```

### Platform

LINUX 5.19.0-43-generic

### Description

```rust
#[debug_handler]
async fn add_web(Form(input): Form<FormAddWeb>) -> impl IntoResponse {
    if let Ok(url) = Url::parse(&input.url.unwrap_or_default()) {
        add_url(url.as_str());
        let target = format!("/web?url={}", url.as_str());
        Redirect::to(&target).into_response()
    } else {
        error!("Invalid url");
        let r = (StatusCode::BAD_REQUEST, "invalid url").into_response();
        dbg!(&r);
        r
    };
}
```

```
[src/main.rs:168] &r = Response {
    status: 400,
    version: HTTP/1.1,
    headers: {
        "content-type": "text/plain; charset=utf-8",
    },
    body: UnsyncBoxBody,
}
```

I have checked <https://docs.rs/axum/latest/axum/response/index.html#regarding-impl-intoresponse> and I know **`impl IntoResponse` can only be used to return a single type**.

But everything looks find until the `else` part return nothing(browser get `ok 200`)

**Comment by davidpdrsn** [comment]
Please elaborate. I don't understand what your issue is.

**Reply by freedit-dev** [reply]
The response should be `status 400...`, but 200 ok instead.

**Reply by davidpdrsn** [reply, marked as answer]
Ah I see it. You have a `;` after the else so you're returning `()`. Another reason we don't recommend returning `impl IntoResponse`

![Untitled](https://github.com/tokio-rs/axum/assets/718941/91338d7e-759a-45ca-8864-1d5e54728c85)
